src/nbody_simulation_result.jl

```diff
@@ -0,0 +1,161 @@
+# SimulationResult sould provide an interface for working with properties of a separate particle
+# and with physical properties of the whole system.
+struct SimulationResult{sType <: NBodySystem}
+    solution::AbstractTimeseriesSolution
+    simulation::NBodySimulation{sType}
+end
+
+function Base.show(stream::IO, sr::SimulationResult)
+    print(stream, "N: ") 
+    show(stream, length(sr.simulation.system.bodies))
+    print("\n")
+    show(stream, sr.simulation.system)
+    print(stream, "Time steps: ") 
+    show(stream, length(sr.solution.t))
+    print("\n")
+    print(stream, "t: ") 
+    show(stream, sr.solution.t)
+end
+
+(sr::SimulationResult)(t) = return sr.solution(t)
+
+# Instead of treating NBodySimulation as a DiffEq problem and passing it into a solve method
+# it is better to use a specific function for n-body simulations.
+function run_simulation(s::NBodySimulation, alg_type=Tsit5(), args...; kwargs...)
+    solution = solve(ODEProblem(s), alg_type, args...; kwargs...)
+    return SimulationResult(solution, s)
+end
+
+# this should be a method for integrators designed for the SecondOrderODEProblem (It is worth somehow to sort them from other algorithms)
+function run_simulation(s::NBodySimulation, alg_type::Union{VelocityVerlet,DPRKN6,Yoshida6}, args...; kwargs...)
```

Comment:
There are too many algorithms in JuliaDiffEq) It is cool, but I am not sure which of them would be good for N-Body simulations. VelocityVerlet is the most mentioned one. I am just not acquainted with others yet, that is why I've chosen these three for a while.

Reply:
Don't choose any. Just leave it open without a type parameter. One of the interesting things will be to test different algorithms. This has no effect on performance anyways.

Reply:
In fact, I put some algorithms here in order to distinguish when to transform an N-body-system into ODEProblem or into SecondOrderODEProblem. Via the multiple dispatch mechanism. I could group all the appropriate algorithms for SecondOrderODEProblem under a particular parent, though I am not sure if it is ok for the current structure of JuliaDiffEq. Another approach is to add an argument like `transformaton_order::Integer = 1`.